server/src/utils.ts

```diff
@@ -39,14 +39,14 @@ export let findProjectRootOfFile = (
 };
 
 export let findBinaryFromProjectRoot = (
-  source: p.DocumentUri
+  projectRootPath: p.DocumentUri // This must be a directory and not a file!
 ): null | p.DocumentUri => {
-  let dir = path.dirname(source);
-  let bscNativeReScriptPath = path.join(dir, c.nodeModulesBinDir);
+  let dir = path.dirname(projectRootPath);
```

Comment:
The style mistake is still there: `dir` is defined too early. It's very easy to change this and introduce a bug again.